applications/asset_tracker/src/cloud_codec/cloud_codec.c

```diff
@@ -83,3 +83,42 @@ int cloud_encode_sensor_data(const struct cloud_sensor_data *const sensor,
 
 	return 0;
 }
+
+int cloud_encode_digital_twin_data(const struct cloud_sensor_data *sensor,
+				 struct cloud_data *output)
+{
+	int ret;
+	char *buffer;
+
+	__ASSERT_NO_MSG(sensor != NULL);
+	__ASSERT_NO_MSG(sensor->data.buf != NULL);
+	__ASSERT_NO_MSG(sensor->data.len != 0);
+	__ASSERT_NO_MSG(output != NULL);
+
+	cJSON *root_obj = cJSON_CreateObject();
+	cJSON *state_obj = cJSON_CreateObject();
+	cJSON *reported_obj = cJSON_CreateObject();
+
+	if (root_obj == NULL) {
```

Comment:
Same story as it was in other function - you should check all objects allocated and release them in case of failure.

Reply:
Added

Reply:
I think it's still unaddressed?

Reply:
Ah yes, my mistake, sorry. Let me try again.